test/framework/resources/templates/default-shoot.yaml

```diff
@@ -28,4 +28,6 @@ spec:
     kubernetesDashboard:
       enabled: true
       authenticationMode: token
+  tolerations:
+  - key: seed.gardener.cloud/protected
```

Comment:
@schrodit Is this template used in the usual create shoot tests?
If so, should we avoid to tolerate protected seeds?
There might be protected seeds in the system for development or testing purposes and we don't necessarily want to schedule TM shoots on them.

What about using a taint specific to the current test run to protect test Seeds, so that only shoots are scheduled on them if really intended?

Reply:
Good catch! Yes that template is used for the test shoot creations and we certainly don't want test shoots generally to be scheduled on to some private test seeds on dev or staging for example.
So you'd suggest to taint a test seed with `testrun-123` and then adapt the shoot creation to tolerate `testrun-123`? Sounds like a good idea to me.

Reply:
Yep, something like this, is what I had in mind.
Probably [`TM_TESTRUN_ID`](https://github.com/gardener/test-infra/blob/f11e8da17be3d35447b4498d7763ac61056bcfaf/pkg/testmachinery/config.go#L72) can be used for that purpose by the managed seed creation and shoot creation to add taints and tolerations.

Reply:
This means I should change the `create-shoot` integration test, adding an additional parameter to pass custom tolerations, correct? And then I should pass a toleration based on `$TM_TESTRUN_ID` in `CreateShoot.yaml`? And similarly for the `create-managed-seed` test?

Reply:
Yes, sounds a bit like overhead, but we really need to avoid scheduling test shoots onto some probably broken test seeds. Unless @schrodit has a better idea?

Reply:
> adding an additional parameter to pass custom tolerations, correct?

I don't think, we need an extra option for passing custom tolerations for now.
We can probably just add a toleration `test.gardener.cloud/test-run=$TM_TESTRUN_ID` (or a similar key) to every shoot created by any integration test, as it doesn't hurt.
In the managed seed test we could then add a corresponding taint (also no custom one needed for now) in any case.
This would basically make the managed seed a "private seed" for just this TestRun.

If we create a `ManagedSeed` and some shoots in the same TestRun they would be able to be scheduled there.
Though, if we want to use the `ManagedSeed` in some other TestRun, we would need to think of a different/shared taint. But I assume, this wouldn't be required for now.

WDYT?

Reply:
Yes, makes sense. Will do as suggested.

Reply:
Nice idea, and I even implemented it, but when I tried to actually execute the integration test I got the following error:
```
time="2021-02-23T15:58:59Z" level=fatal msg="Cannot create shoot tm-x93-ruf-seed: shoots.core.gardener.cloud \"tm-x93-ruf-seed\" is forbidden: error while validating tolerations against allowlist: [spec.tolerations[0]: Forbidden: only the following tolerations are allowed: [seed.gardener.cloud/protected]]"
```
@timebertt Do you think it's ok to add `test.gardener.cloud/test-run` to the allowlist above?

Reply:
I would even say, we could always allow this taint. It's a well-known taint key which is reserved for Integration Tests, so no one should use it for anything other than that, which makes it harmless to add / always allow.

Reply:
@dguendisch Will take care, so I consider this done as far as this PR is concerned.